src/nnet3/nnet-simple-component.cc

```diff
@@ -392,6 +430,19 @@ void NormalizeComponent::Backprop(const std::string &debug_info,
   in_deriv->AddDiagVecMat(-1.0 / d_scaled,
                           dot_products, in_value,
                           kNoTrans, 1.0);
+
+  if (add_log_stddev_) {
```

Comment:
You wouldn't need to use a temporary matrix if you computed this before the other part of the input derivative.